galley/pkg/config/analysis/local/analyze.go

```diff
@@ -101,11 +114,49 @@ func (sa *SourceAnalyzer) AddFileKubeSource(files []string, defaultNs string) er
 
 // AddRunningKubeSource adds a source based on a running k8s cluster to the current SourceAnalyzer
 func (sa *SourceAnalyzer) AddRunningKubeSource(k client.Interfaces) {
+	// As an optimization, filter out the resources we won't need for the current analysis.
+	// This matters because getting a snapshot from k8s is relatively time-expensive,
+	// so removing unnecessary resources makes a useful difference.
+	filteredResources := make([]schema.KubeResource, 0)
+	for _, r := range sa.m.KubeSource().Resources() {
+		if _, ok := sa.collectionsUsed[r.Collection.Name]; !ok {
+			scope.Analysis.Debugf("Disabling resource %q since it isn't necessary for the current analysis", r.Collection.Name)
+			r.Disabled = true
+		}
+
+		filteredResources = append(filteredResources, r)
+	}
+
 	o := apiserver.Options{
 		Client:    k,
-		Resources: sa.m.KubeSource().Resources(),
+		Resources: filteredResources,
 	}
-	src := apiserver.New(o)
+	src := apiserverNew(o)
 
 	sa.sources = append(sa.sources, src)
 }
+
+func getUpstreamCollections(analyzer analysis.Analyzer, xformProviders transformer.Providers) map[collection.Name]bool {
```

Comment:
FYI, I'll probably refactor this code and hoist the core logic to schema, so that the same logic can also be used by Galley itself.

Reply:
👍